### PR TITLE
Use monitor smart pointers

### PR DESCRIPTION
**Pull request by WhySoBad**
Hi,

It's been some time since vaxry broke the plugin. In the latest git release `v0.44.0-63-g08cc063e` the API now returns `CSharedPointer<CMonitor>` everywhere instead of `CMonitor*` as it used to. 

I've seen the plugin already handled some cases with smart pointers by getting the inner pointer hand operating on it directly. I've now switched the whole monitor logic to use `CSharedPointer<CMonitor>` since I think it's better to change it everywhere now as the Hyprland API progresses to replace all pointers by smart pointers.

Additionally, I've applied some suggestions provided by clangd (e.g. the use of `const auto&` references etc.)

### CODE REVIEW

**Comment by levnikmyskin** [comment]
Hey! Thank you so much for your effort, as usual. I'll look into this PR more in depth tomorrow, but looks good as far as I can see.
I never liked this unordered map thing tbh, dunno why I implemented like this back then, and now it seems like it's bringing in some more complexit. I think this could just be a vector, but maybe I'm not remembering something. Anyway, nothing really related to this PR.

File: src/VirtualDeskManager.cpp [hunk 1]
```diff
@@ -280,16 +280,16 @@ void VirtualDeskManager::invalidateAllLayouts() {
...
     if (currentMonitor && (!currentMonitor->m_bEnabled || !currentMonitor->output)) {
         for (auto mon : g_pCompositor->m_vMonitors) {
             if (mon->m_bEnabled && mon->output)
-                return mon.get();
+                return mon;
         }
         return nullptr;
     }
-    return currentMonitor;
+    return currentMonitor.lock();
```
Comment (levnikmyskin) [inline src/VirtualDeskManager.cpp:294]: what does this do btw?

**Review by levnikmyskin — APPROVED** [review]
LGTM, just got one comment, but feel free to merge whenever you want

File: src/VirtualDeskManager.cpp (hunk 1 above)
Comment (WhySoBad) [reply src/VirtualDeskManager.cpp:294]: it's used to convert the weak pointer returned by the hyprland api to a shared pointer

**Pull request merged**